### PR TITLE
Abductors can now make custom abductee objectives

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

This allows abductors to set custom abductee objectives, rather than needing to stack a brainwash on top of the dumb rng objective. It's properly logged, and will ofc fall back to random objectives if a custom objective is not set.

## Why It's Good For The Game

Encourages creativity with abductors, allowing them to make people go crazy in unique and personalized ways, without shitty rng objectives getting in the way of them doing so.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

23-07-07-1688753443-dreamseeker
23-07-07-1688753479-dreamseeker

</details>

## Changelog
:cl:
add: Abductors can now imprint custom objectives onto the psyche of their victims whenever processing them through their experimenter machine!
add: Abductee objectives can be cured, by lobotomizing the victim after they've had their abductor gland removed.
/:cl:

### CODE REVIEW

**Comment by LodedDiper69** [comment]
Cool, but what will stop them from being uncreative and going "help the abductors" or something?

**Comment by St0rmC4st3r** [comment]
> Cool, but what will stop them from being uncreative and going "help the abductors" or something?

Administrative issue

**Comment by EvilDragonfiend** [comment]
I think mindshielded people shouldn't get customised brainwash objects by this.

**Comment by Rukofamicom** [comment]
I think mind shielded people should be immune to abductor brainwashing altogether.

**Comment by Isy232** [comment]
Agree with Ruko, abductor brainwashing is specifically weird in that it wont mess with a mindshield.

Maybe it should automatically add a clause that says "you should attempt to avoid being mindshielded" as most of the time brainwashing doesn't include such a clause and it becomes a grey area of if your allowed to accept being remindshielded or not.

**Comment by Absolucy** [comment]
tbh I always presumed that abductors were just advanced enough where they didn't really care about mindshields anyways

File: code/modules/antagonists/abductor/abductor.dm
Comment (PowerfulBacon) [inline code/modules/antagonists/abductor/abductor.dm:199]: Crash if custom objective is null rather than safely handle it in a way that causes future bugs without a stack trace

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Send an admin message every time a custom objective is sent. Also note that all the objectives will just be set to help the abductors rather than do anything interesting.

File: code/modules/antagonists/abductor/abductor.dm
Comment (Absolucy) [reply code/modules/antagonists/abductor/abductor.dm:199]: But if `custom_objective` is null... it'll just do a random objective instead?

Comment (PowerfulBacon) [reply code/modules/antagonists/abductor/abductor.dm:199]: Ah yea, I see. This is the antagonist and not an objective

**Comment by Absolucy** [comment]
It _does_ message admins every time a custom objective is used
https://github.com/BeeStation/BeeStation-Hornet/pull/8996/files#diff-b65239bebfc74e3a14e8af1c261e5ce4c9a4aaf10afd0a95ea3e4c833c8167d6R150-R154

File: code/modules/antagonists/abductor/machinery/experiment.dm
Comment (EvilDragonfiend) [inline code/modules/antagonists/abductor/machinery/experiment.dm:117]: put `if(OOC_FILTER_CHECK(custom_objective))` here and nullify the message and send admin report if it's caught.
So that'll be a random abductee brainwash instead.

**Review by EvilDragonfiend — APPROVED** [review]
* code is fine
* consulted with an admin and I hope this will be fine after I talked to them

It's conditional approval. It needs to have a OOC filter to check the message.

**Comment by EvilDragonfiend** [comment]
actually might need more opinions from just one admin.

**Comment by Tyranicranger4** [comment]
>> Cool, but what will stop them from being uncreative and going "help the abductors" or something?

>Administrative issue

That doesn't change that there's absolutely no reason why they should be able to give people a customizable and irreversible objective. Even if it's being used somewhat responsibly that's just bad. It needs to be reversible in some way, similarly to how actual brainwashing functions.

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
I like custom objectives a lot more than current objectives. I think it will introduce some problems, but the potential to improve the abductee experience (which is generally very poor right now) is worth the new potential problems. 

I do agree that the brainwashing should be reversible through means more accessible than alien research however, especially given the new issues custom objectives may introduce.

**Comment by Absolucy** [comment]
@Rukofamicom @Tyranicranger4 I'm confused at this point, there's several conflicting messages I'm receiving

1. I previously made a PR that made lobotomies cure abductee objectives, and was told it was a bad idea due to the fact people would just be lobotomized any time they got abducted.
2. But... now, abductee objectives should be reversible? Wouldn't this result in the same issue as 1?

I'm not sure how to reconcile those two things in an acceptable manner.

**Comment by Rukofamicom** [comment]
> I previously made a PR that made lobotomies cure abductee objectives, and was told it was a bad idea due to the fact people would just be lobotomized any time they got abducted.

The PR was closed the day after it was opened with minimal discussion. First impressions don't always indicate what the maintainers or even server as a whole want - it can easily mean one person had strong feelings and shared it in a chat with friends. I went into this discussion not knowing there was no way to fix abductor brainwashing currently, and I wasn't alone on the backside of the discussion. It makes no sense that there's no way to reverse what they do in the code.

There's more nuance with custom objectives as opposed to the nothing-burger objectives as well - the crew will very likely **need** the ability to reverse these, as custom objectives will easily seize permanent control of a round otherwise.

**Review by PowerfulBacon — APPROVED** [review]

**Comment by Absolucy** [comment]
@Rukofamicom I've made it so, if you do a lobotomy after removing the abductor gland, it will remove an abductee custom objective (although it will _not_ remove the antag datum, nor will it remove non-custom abductee objectives).

**Comment by EvilDragonfiend** [comment]
I won't merge this myself because I believe this PR will be good with my organ change (which I can't do selfmerge with this)
* https://github.com/BeeStation/BeeStation-Hornet/pull/9097

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
Final request is to allow the lobotomy to cure any abductor objectives, not just custom so there is uniform consistency.

Send a discord ping when this is done.

**Comment by Rukofamicom** [comment]
There is some concern the lobotomy's downsides might make the cost of "curing" too high, but we'd like to give it a shot rather than prematurely jumping straight into a no-cost or lower cost cure.

Looking forward to what people do with custom objectives.

**Comment by Absolucy** [comment]
23-07-07-1688753443-dreamseeker
23-07-07-1688753479-dreamseeker

**Review by Rukofamicom — APPROVED** [review]
Looks good at a (phone) glance

**Pull request merged**
